src/NavigationActions.js

```diff
@@ -11,10 +11,15 @@ const RESET = 'Navigation/RESET';
 const SET_PARAMS = 'Navigation/SET_PARAMS';
 const URI = 'Navigation/URI';
 
-const createAction = (type: string) => (payload: Object = {}) => ({
-  type,
-  ...payload,
-});
+const createAction = (type: string) => {
+  const fn = (payload: Object = {}) => ({
+    type,
+    ...payload,
+  });
+
+  (fn: any).toString = () => type;
```

Comment:
The type-checker throws the following error.

```
src/NavigationActions.js:20
 20:   (fn: mixed).toString = () => type;
                   ^^^^^^^^ property `toString`. Property cannot be assigned on
 20:   (fn: mixed).toString = () => type;
            ^^^^^ mixed
```

To my understanding, the `mixed` type is used to represent a function with arbitrary argument and return types. The argument/return types of the function are a separate concern from assignment to a property on the function.

I think the only valid typecast here would be to a `Stringifyable` interface that defines a `toString` field. However, I don't believe something like that exists out-of-the-box, and it seems like overkill to add it for this use case.